systemtest/040-local-registry-auth.bats

```diff
@@ -111,7 +112,8 @@ function setup() {
 }
 
 teardown() {
-    podman rm -f reg
+    # Need to change XDG_RUNTIME_DIR when podman removes the registry.
+    XDG_RUNTIME_DIR=$XDG_RUNTIME_DIR_OLD podman rm -f reg
```

Comment:
This does not do what you think it does. This runs the `podman rm` command with the old XDG... but leaves XDG the way it is. Then, on the next `@test`, and next `setup()`, `OLD` will be set to the current `XDG`, which is probably not what you want. 

I don't remember this code, and have not fully read it, but I believe the correct approach would be:
```bash
setup() {
    # temporarily override XDG for (whatever reason) but preserve old setting
    export OLD=$XDG
    export XDG=$newvalue
}
....
teardown() {
   # restore XDG
   XDG=$OLD
```